File: scripts/src/contracts/sandbox/schemas.ts
import { BorshSchema, Infer as BorshInfer } from "borsher";

export const schemaWordCountRequest = BorshSchema.Struct({
  data: BorshSchema.String,
});

export const schemaInscribeWordCountRequest = BorshSchema.Struct({
  inscribe: BorshSchema.String,
});

export type IWordCountRequest = BorshInfer<typeof schemaWordCountRequest>;

export const schemaWordCountResponse = BorshSchema.Struct({
  calldata_echo: BorshSchema.String,
  inscribe_echo: BorshSchema.String,
  calldata_count: BorshSchema.u16,
  inscribe_count: BorshSchema.u16,
});

export type IWordCountResponse = BorshInfer<typeof schemaWordCountResponse>;

File: contracts/sandbox/src/schemas.rs
use borsh::{BorshDeserialize, BorshSerialize};

#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
pub struct BorshWordCountRequest {
    pub data: String,
}
#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
pub struct BorshWordCountResponse {
    pub calldata_echo: String,
    pub inscribe_echo: String,
    pub calldata_count: u16,
    pub inscribe_count: u16,
}

#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
pub struct BorshWordCountInscribeRequest {
    pub inscribe: String,
}
